### DISCUSSION TITLE
show docstring of __call__ if variable is assigned to a type instance and the type has __call__ defined unless variable itself has its own doc string

### DISCUSSION THREAD

**Discussion by heejaechang** [microsoft/pylance-release#2917] [Enhancement]
variable = SomeType()

when user hover variable, currently we don't show anything, but it would be nice if it shows doc string in this order

show docstring of variable if it is defined

ex) variable = ...
""" doc string ... "

if not, see whether variable is type alias or regular variable

if it is type alias
ex) variable = SomeType

show class doc string if defined
ex) class SomeType:
""" doc string"""

if not, show docstring of ```__init__``` if defined
```
class SomeType:
     def __init__(..):
          """ doc string """
```
if not, show doscstring of sometype metaclass's ```__call__`` if defined
```
class Meta(...):
    def __call__(..):
        """ doc string """

class SomeType(metaclass=Meta):
    ...

variable = SomeType
```

if it is a variable
ex) variable = SomeType()

show docstring of ```__call__``` if defined

```
class SomeType():
      def __call__(...):
           """ doc string """
```

also, in case there is ```__call__``` or if it is type alias, show variable as function or actual type (function or class) it is aliased, rather than showing it as a variable.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
